irods/session.py

```diff
@@ -47,7 +49,8 @@ def __exit__(self, exc_type, exc_value, traceback):
 
     def __del__(self):
         self.do_configure = {}
-        self.cleanup()
+        if self.pool is not None:
```

Comment:
consider a comment for 'why' here (cover the same ground as in the commit message)

could save the next reader some questions.

Reply:
ah ok...

Reply:
Seems fine to me. @trel?

Reply:
yes, thanks.